plugins/outputs/nats/nats.go

```diff
@@ -85,8 +132,51 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if err != nil {
+		return err
+	}
+
+	if n.Jetstream != nil {
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+
+		if len(n.Jetstream.Subjects) == 0 {
+			n.Jetstream.Subjects = []string{n.Subject}
+		}
+		if !choice.Contains(n.Subject, n.Jetstream.Subjects) {
+			n.Jetstream.Subjects = append(n.Jetstream.Subjects, n.Subject)
+		}
+		var streamConfig jetstream.StreamConfig
+		n.convertToJetstreamConfig(&streamConfig)
+		_, err = n.jetstreamClient.CreateOrUpdateStream(context.Background(), streamConfig)
+		if err != nil {
+			return fmt.Errorf("failed to create or update stream: %w", err)
+		}
+		n.Log.Infof("stream (%s) successfully created or updated", n.Jetstream.Name)
+	}
+	return nil
+}
+
+func (n *NATS) convertToJetstreamConfig(streamConfig *jetstream.StreamConfig) {
+	telegrafStreamConfig := reflect.ValueOf(n.Jetstream).Elem()
+	natsStreamConfig := reflect.ValueOf(streamConfig).Elem()
+	for i := 0; i < telegrafStreamConfig.NumField(); i++ {
+		destField := natsStreamConfig.FieldByName(telegrafStreamConfig.Type().Field(i).Name)
+		if destField.IsValid() && destField.CanSet() {
+			destField.Set(telegrafStreamConfig.Field(i))
+		}
+	}
+}
```

Comment:
This has footgun potential as you circumvent Golangs type-checking. If for example one field is removed upstream (or the type is changed) you silently will ignore the Telegraf setting and nobody will notice until some unexpected behavior happens. I really would feel better if we explicitly set the config-fields here (even though this bloats the code) **or** if we use the `jetstream.StreamConfig` directly in the Telegraf configuration.

Reply:
Using `jetstream.StreamConfig` directly was done before, but rejected since it didn't have toml conversion, but only json.

Reply:
Well it is not absolutely necessary to have `toml` tags, but I agree and would prefer the explicit setting of options. The question is, do we really need to expose all those options or can we live with the defaults?

Reply:
Yes it is not necessary to have toml tags. but there are some fields which did not support toml unmarshalling such as retention, limits. This is the reason why in the configuration, users are expected to pass integer values for such fields because it we can resuse the Stringer interface defined on those types. 

There are use cases when we definitely need full set of options.

Reply:
@neelayu ok, so we can go with the full set of options, but I really think we should not hamper the user experience with the integers just because it's more convenient to use the String() interface. So please use speaking-strings where possible and do the explicit transformation. @powersj what do you think?

Reply:
Will make the changes.